contracts/core/extensions/CoreIssuanceOrder.sol

```diff
@@ -268,17 +269,18 @@ contract CoreIssuanceOrder is
             // Verify exchange address is registered
             require(exchange != address(0));
 
-            // Read the order body based on header order length info
-            uint256 exchangeDataLength = header.totalOrdersLength.add(160);
+            // Read the order body based on order data length info in header plus the length of the header (128)
+            uint256 exchangeDataLength = header.orderDataBytesLength.add(128);
             bytes memory bodyData = LibBytes.slice(
                 _orderData,
-                scannedBytes.add(160),
+                scannedBytes.add(128),
                 scannedBytes.add(exchangeDataLength)
             );
 
-            // Transfer header.makerTokenAmount to Exchange Wrapper
+            // Transfer maker token to Exchange Wrapper to execute exchange orders
+            // Using maker token from signed issuance order to prevent malicious encoding of another maker token
             ITransferProxy(state.transferProxy).transfer(
-                header.makerTokenAddress,
+                _makerTokenAddress,
                 header.makerTokenAmount,
```

Comment:
The exchange header still specifies how much of the maker token to send to each wrapper to execute the orders/trades, but it must be of the maker token from the signed issuance order. If it is the wrong erc20 token, the wrapper will not have enough to execute the trades.